docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+which means that a successful cluster registration requires both sides of approval 
+and commitment from the hub cluster and the managed cluster.
+
+The **work agent** monitors the `ManifestWork` resource in the cluster namespace 
+on the hub cluster. The work agent tracks all the resources defined in ManifestWork 
+and updates its status. There are two types of status in ManifestWork: the *resourceStatus* 
+tracks the status of each manifest in the ManifestWork, and *conditions* reflects the overall 
+status of the ManifestWork. The work agent checks whether a resource is *Available*, 
+meaning the resource exists on the managed cluster, and *Applied*, meaning the resource 
+defined in ManifestWork has been applied to the managed cluster. To ensure the resources 
+applied by ManifestWork are reliably recorded, the work agent creates an *AppliedManifestWork* 
+on the managed cluster for each ManifestWork as an anchor for resources relating to ManifestWork. 
+When ManifestWork is deleted, the work agent runs a *Foreground* deletion, and that ManifestWork 
+will stay in deleting state until all its related resources have been fully cleaned in the managed 
+cluster.
+
+## OCM Status Add-On Agent
+
+The OCM Status Add-On Agent is a controller that runs alongside the OCM Agent 
+in the managed cluster. Its primary function is to track objects delivered 
+by the work agent and report the full status of those objects back to the ITS. 
+The KubeStellar controller can then use different user-defined summarization 
+policies to report status, such as the `wantSingletonReportedState` policy that reports 
+full status for each deployed object when the workload is delivered only to one 
+cluster. The controller watches *appliedmanifestworks* objects to determine which 
+objects have been delivered through the work agent. It then starts dynamic informers 
+to watch those objects, collect their individual statuses, and report back the status 
+updating *workstatuses* objects in the namespace associated with the WEC in the ITS.
+
+## KubeStellar Controllers Architecture
+
+The KubeStellar controllers architecture is based on common patterns and best 
+practices for Kubernetes controllers, such as the 
+[<u>Kubernetes Sample Controller</u>](https://github.com/kubernetes/sample-controller). 
+A Kubernetes controller uses informers to watch for changes in Kubernetes
+objects, caches to store the objects, event handlers to react to
+events, work queues for parallel processing of tasks, and a reconciler
+to ensure the actual state matches the desired state. However, that
+pattern has been extended to provide the following features:
+
+- Using dynamic informers
+- Starting informers on all API Resources (except some that do not need
+  watching)
+- Informers and Listers references are maintained in a hash map and
+  indexed by GVK (Group, Version, Kind) of the watched objects.
+- Using a common work queue and set of workers, where the key is defined as follows:
```

Comment:
FYI: There are lots of kinds of keys floating around. I generally use the term "object reference" here, to make it clearer what these things are.